pyfranca/tests/test_franca_processor.py

```diff
@@ -665,3 +665,4 @@ def test_import_file_chain(self):
             }
         """)
         self.processor.import_file(fspec)
+        self.processor.import_file("./Type1.fidl")
```

Comment:
Why are you explicitly importing `Type1.fidl` here? It is already referenced by `P.fidl`.

I do not understand the intention.

Reply:
Is it correct that the intention is to reference `Type1.fidl` two times with different paths and the expectation is that `Processor` will detect this and ignore the second reference?

Reply:
**Why are you explicitly importing Type1.fidl here? It is already referenced by P.fidl.**
You are right. But you expect that the code generator knows which fidl file is the head. My Generator only knows the directory and take all fidl files in this directory and import them in the processor. So it could be that a fidl file is imported twice with different paths in parameter fspec. 

This is the reason why I found this "bug".  ;-)

**Is it correct that the intention is to reference Type1.fidl two times with different paths and the expectation is that Processor will detect this and ignore the second reference?**
yes 

I think my solution make the file import more robust against misuse. We do not realy know how the pyfranca user would call import_file(). Thats why I think it is a good approach to normalize user inputs before using.